scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/Record.java

```diff
@@ -447,27 +447,31 @@ private Record(ProtocolVersion version, int epoch, long sequenceNumber) {
 		CipherSuite cipherSuite = currentReadState.getCipherSuite();
 		LOGGER.trace("Decrypting record fragment using current read state{}{}", StringUtil.lineSeparator(), currentReadState);
 
-		switch (cipherSuite.getCipherType()) {
-		case NULL:
-			// do nothing
-			break;
-			
-		case AEAD:
-			result = decryptAEAD(ciphertextFragment, currentReadState);
-			break;
-			
-		case BLOCK:
-			result = decryptBlockCipher(ciphertextFragment, currentReadState);
-			break;
-			
-		case STREAM:
-			// Currently, Scandium does not support any stream ciphers
-			// RC4 is explicitly ruled out from being used in DTLS
-			// see http://tools.ietf.org/html/rfc6347#section-4.1.2.2
-			break;
+		try {
+			switch (cipherSuite.getCipherType()) {
+			case NULL:
+				// do nothing
+				break;
 
-		default:
-			break;
+			case AEAD:
+				result = decryptAEAD(ciphertextFragment, currentReadState);
+				break;
+
+			case BLOCK:
+				result = decryptBlockCipher(ciphertextFragment, currentReadState);
+				break;
+
+			case STREAM:
+				// Currently, Scandium does not support any stream ciphers
+				// RC4 is explicitly ruled out from being used in DTLS
+				// see http://tools.ietf.org/html/rfc6347#section-4.1.2.2
+				break;
+
+			default:
+				break;
+			}
+		} catch(RuntimeException ex) {
+			throw new GeneralSecurityException(ex.getMessage(), ex);
```

Comment:
Should we add a LOG with warn because this should not happened ?

I mean the idea would be to only raise "GeneralSecurityException" for case we understand (anticipated case).
A runtime would be an unexpected error, we don't want this error terminate the connection but we want to know when this happened to fix the code and  handle it in a more proper way.

Reply:
I added an extensive test for malformed records. 
Therefore I removed the exception conversion.